app.py

```
@@ -232,6 +232,31 @@ def wait_for_quorum(replica_list, main_url):
     logger.info("Quorum has been established with {} as the leader", main_url)
 
 
+def get_vault_pods():
+    if pod_retrieval_max_retries <= 0:
+        print("Pod retrieval max retries cannot be lower than 1: {}", pod_retrieval_max_retries)
+        exit(2)
+
+    for _ in range(pod_retrieval_max_retries):
+        pod_list = api_instance.list_namespaced_pod(
+            namespace=vault_namespace, label_selector="vault-sealed=true"
+        )
+
+        if any(p.status.pod_ip is None for p in pod_list.items):
+            sleep(scan_delay)
+            continue
+
+        return pod_list
+
+    if len(pod_list.items) == 0:
+        print("Not Vault pods found")
+        exit(2)
+
+    print("Vault pods have no assigned IP address: {}",
+          [p.metadata.name for p in pod_list.items if p.status.pod_ip is None])
+    exit(2)
```

Are you sure for exit 2? 
in this case, pod will restart

---

I was thinking the best here would be to restart pod so it goes into a CrashLoopBackoff, so people check the logs to see what the issue was. How do you handle the cases where there is nothing to unseal or the vault pods are not ready? Do you want to silently fail, or do you agree with my proposal?

---

I agree, please add logging for exit reason.

---

Is using `logger` enough or do you want me to `print` the message?

---

Please use logger

---

That's already implemented. Could you please review the code again?